external-crates/move/crates/move-vm-runtime/src/jit/execution/translate.rs

```diff
@@ -203,76 +204,93 @@ fn module(
     dbg_println!("Loading module: {}", self_id);
 
     // Load module types
-    load_module_types(
+    let (datatype_descriptors, descriptor_map) = load_module_types(
         package_context,
         link_context,
         package_context.runtime_id,
         package_id,
         &module.compiled_module,
     )?;
+
+    let cmodule = &module.compiled_module;
+
+    // NB: Note that this assumes the datatype descriptors will live at least as long as the
+    // vtable -- if that is not true, we _will_ segfault.
+    record_module_types_in_vtable(&mut package_context.vtable, &descriptor_map);
+
     dbg_println!("Module types loaded");
 
     // Initialize module data
-    let type_refs = initialize_type_refs(&module.compiled_module)?;
+    let type_refs = initialize_type_refs(package_context, &cmodule)?;
 
-    let mut instantiation_signatures: BTreeMap<SignatureIndex, Vec<Type>> = BTreeMap::new();
+    let (instantiation_signatures, _signature_map) = cache_signatures(package_context, &cmodule)?;
```

Comment:
Can remove `_signature_map`? Or is there a plan to use/remove in a future PR?

Reply:
We should not -- removed.